cache.go

```diff
@@ -0,0 +1,99 @@
+package wazero
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	goruntime "runtime"
+
+	"github.com/tetratelabs/wazero/api"
+	"github.com/tetratelabs/wazero/internal/compilationcache"
+	"github.com/tetratelabs/wazero/internal/version"
+	"github.com/tetratelabs/wazero/internal/wasm"
+)
+
+// Cache is the configuration for caching behavior of wazero.Runtime and can be passed to wazero.RuntimeConfig.
+type Cache interface {
+	// Closer can be used to clear the in-memory cache and this doesn't delete this cache directory when configured.
```

Comment:
I think this isn't very clear - maybe better to have separate Close APIs depending on deleting the file caches or not.

Reply:
having said that, given that so far no one has requested delete file behavior on the experimental.CompilationCache API, so maybe this is just fine

Reply:
do we expect this to be used for anything except compilation? If not, I think we should say CompileCache, as it matched CompileModule (or something like that). then when you link the godoc together it seems more intuitive